core/modules/layout/layout.admin.inc

```diff
@@ -1005,8 +997,48 @@ function layout_block_add_page(Layout $layout, $renderer_name, $region_name, $bl
             'data-dialog-options' => json_encode(array('dialogClass' => 'layout-dialog')),
           )
         )
-      ),
-      '#description' => isset($block_info['description']) ? $block_info['description'] : NULL,
+      );
+      $description = isset($block_info['description']) ? $block_info['description'] : NULL;
+    }
+    else {
+      // Remove the main page content block since it's special.
+      if ($block_info['info'] == 'Main page content') {
+        continue;
+      }
+
+      $link = $block_info['info'];
+      $description = isset($block_info['description']) ? $block_info['description'] . ' ': '';
+
+      if (count($required_contexts) == 1) {
+        $required_context = reset($required_contexts);
+        switch($required_context) {
+          case 'node':
+            $requirements_info = t('<strong>Requires:</strong> A layout with a <em>node</em> context available, for example: on the path node/%.');
+            break;
+          case 'term':
+            $requirements_info = t('<strong>Requires:</strong> A layout with a <em>user</em> context available, for example: on the path user/%.');
+            break;
+          case 'term':
+            $requirements_info = t('<strong>Requires:</strong> A layout with a <em>term</em> context available, for example: on the path taxonomy/term/%.');
+            break;
+            // Unknown path, not safe to use as an example.
+          default:
+            $requirements_info = t('<strong>Requires:</strong> A layout with a <em>!context</em> context available.', array('!context' => $required_context));;
+            break;
```

Comment:
All of these strings can (and should) be automated by using `layout_get_context_info($required_context)`, then pulling off the first `menu paths` value to use as an example.